test/Search_UnitTestHelpers.hpp

```diff
@@ -102,8 +102,23 @@ auto query(Tree const &tree, Queries const &queries)
       Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, values));
 }
 
-#define ARBORX_TEST_QUERY_TREE(tree, queries, reference)                       \
-  BOOST_TEST(query(tree, queries) == (reference),                              \
+template <typename ExecutionSpace, typename Tree, typename Queries>
+auto query(ExecutionSpace const &exec_space, Tree const &tree,
+           Queries const &queries)
+{
+  using memory_space = typename Tree::memory_space;
+  using value_type =
+      std::conditional_t<is_distributed<Tree>{}, Kokkos::pair<int, int>, int>;
+  Kokkos::View<value_type *, memory_space> values("Testing::values", 0);
+  Kokkos::View<int *, memory_space> offsets("Testing::offsets", 0);
+  tree.query(exec_space, queries, values, offsets);
+  return make_compressed_storage(
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, offsets),
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, values));
+}
```

Comment:
Why did you keep the old overload without the execution space argument (above)

Reply:
It's used in testing by Boost queries. Now that I introduced an update to boost, it can probably be removed.

Reply:
Fixed.